### PR TITLE
A few gm commands and teleport fix

### PR DESCRIPTION
**Pull request by matigramirez**
Added commands:
- `/cure <player_name>`
- `/bsummon <player_name> <x> <z> <map>`
- `/cmove <mapId>`
- `/warning <player_name> "<message>"`: this command doesn't seem to work in ep8. I replicated exactly the same packets as ep4 and it didn't work but the game does send the warning command so I don't really know if it's supposed to work and I'm doing something wrong or not. For the moment, all the logic is already done to fix it in the future if the solution is found.

There was an issue with all teleport commands because the teleport action was being executed twice.
```cs
Teleport(mapId, newPosX, PosY, newPosZ);
_packetsHelper.SendGmTeleport(Client, this);
```
This would send 2 different teleport packets to the client, the normal character teleport packet, and the gm teleport packet, which caused errors on the client; the character would remain in the same map on the game but the server would change it's position and map. This PR solves this issue by adding an extra parameter `bool: teleportedByAdmin` on the `Teleport` method (and the subsequent methods) to ensure that the correct packet is sent.
The error I mention didn't happen always that's why it might have slipped until now, but it hasn't happened again with these changes, at least for me.

### CODE REVIEW

**Comment by bigbadben89** [comment]
The /warning command can be used as an exploit to crash the server and
shouldn't be implemented  unless u can fix it just heads up :) great work
tho look Ford  to testing the files once they more finished

**Comment by matigramirez** [comment]
@bigbadben89 I appreciate the input but this is a server emulator so it can't be used to crash the server as in servers with the original shaiya server files unless the exploit is present in our code aswell.

File: src/Imgeneus.World/Game/Player/CharacterPacketHandlers.cs [hunk 1]
```diff
@@ -550,5 +548,96 @@ private void HandleEnterPortalPacket(CharacterEnteredPortalPacket enterPortalPac
             if (!success)
                 SendPortalTeleportNotAllowed(reason);
         }
+
+        private void HandleGMCreateMob(GMCreateMobPacket gmCreateMobPacket)
+        {
+            if(!_databasePreloader.Mobs.ContainsKey(gmCreateMobPacket.MobId))
```
Comment (aosyatnik) [inline src/Imgeneus.World/Game/Player/CharacterPacketHandlers.cs:554]: I think it makes sense to return here. I.e.
```
if (!_databasePreloader.Mobs.ContainsKey(gmCreateMobPacket.MobId)) {
_packetsHelper.SendGmCommandError(Client, PacketType.GM_CREATE_MOB);
return;
}
```

File: src/Imgeneus.World/Game/Player/CharacterPacketHandlers.cs [hunk 2]
```diff
@@ -550,5 +548,96 @@ private void HandleEnterPortalPacket(CharacterEnteredPortalPacket enterPortalPac
...
+            }
+
+            _packetsHelper.SendGmCommandSuccess(Client);
+        }
+
+        private void HandleGMCurePlayerPacket(GMCurePlayerPacket gmCurePlayerPacket)
+        {
+            var target = _gameWorld.Players.FirstOrDefault(p => p.Value.Name == gmCurePlayerPacket.Name).Value;
+
+            if(target == null)
```
Comment (aosyatnik) [inline src/Imgeneus.World/Game/Player/CharacterPacketHandlers.cs:574]: Here you can also return:
```
if(target == null) {
_packetsHelper.SendGmCommandError(Client, PacketType.GM_CURE_PLAYER);
return;
}
```

File: src/Imgeneus.World/Game/Player/CharacterPacketHandlers.cs [hunk 3]
```diff
@@ -550,5 +548,96 @@ private void HandleEnterPortalPacket(CharacterEnteredPortalPacket enterPortalPac
...
+            target?.FullRecover();
+
+            _packetsHelper.SendGmCommandSuccess(Client);
+        }
+
+        private void HandleGMWarningPlayer(GMWarningPacket gmWarningPacket)
+        {
+            var target = _gameWorld.Players.FirstOrDefault(p => p.Value.Name == gmWarningPacket.Name).Value;
+
+            if(target == null)
```
Comment (aosyatnik) [inline src/Imgeneus.World/Game/Player/CharacterPacketHandlers.cs:586]: and here

File: src/Imgeneus.World/Game/Player/CharacterPacketHandlers.cs [hunk 4]
```diff
@@ -550,5 +548,96 @@ private void HandleEnterPortalPacket(CharacterEnteredPortalPacket enterPortalPac
...
+            Teleport(mapId, newPosX, PosY, newPosZ, true);
+        }
+
+        private void HandleGMTeleportPlayer(GMTeleportPlayerPacket gmTeleportPlayerPacket)
+        {
+            var (name, newPosX, newPosZ, mapId) = gmTeleportPlayerPacket;
+
+            var target = _gameWorld.Players.FirstOrDefault(p => p.Value.Name == name).Value;
+
+            if(target == null)
```
Comment (aosyatnik) [inline src/Imgeneus.World/Game/Player/CharacterPacketHandlers.cs:629]: and here

**Review by aosyatnik — CHANGES_REQUESTED** [review]
@matigramirez overall it looks good. I noticed, that your code is not always formatted, which means, that you are likely not using VS tools for formating. You can format your code with short cuts. But I prefer auto-formater, that formats your code on save, something like [this](https://marketplace.visualstudio.com/items?itemName=mynkow.FormatdocumentonSave).

**Comment by matigramirez** [comment]
@aosyatnik idk why I missed the returns, they are fixed now

**Comment by aosyatnik** [comment]
Looks great! Thanks!

**Review by aosyatnik — APPROVED** [review]

**Pull request merged**